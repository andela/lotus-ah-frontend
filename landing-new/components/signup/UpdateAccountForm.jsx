// react libraries
import React from 'react';

// third party libraries
import PropTypes from 'prop-types';
import Loader from 'react-loader';

// components
import SignupInputGroup from './SignupInputGroup';

/**
 * @desc renders update user detail form
 * @param {string} email
 * @param {function } handleSubmit
 * @return update form
*/
const UpdateAccountForm = ({ email, handleSubmit, loading }) => (
  <form
    className='form-row'
    onSubmit={handleSubmit}
  >
  <div className="col-10 offset-1">
    <div className="input-wrap">
      <SignupInputGroup
        value={email}
        iconClassName='fas fa-envelope'
        name='email'
        type='email'
        placeholder='email'
        inValidfeedback='Please your email'
        disabled={true}
      />
      <SignupInputGroup
        iconClassName='fas fa-user'
        name='fullname'
        type='text'
        placeholder='fullname'
        inValidfeedback='Please enter your full name'
      />
      <SignupInputGroup
        iconClassName='fas fa-at'
        name='username'
        type='text'
        placeholder='user name'
        inValidfeedback='Please enter a username'
      />
        <SignupInputGroup
          iconClassName='fas fa-lock'
          name='password'
          type='password'
          placeholder='password'
          inValidfeedback='Please enter your password.'
        />
         <SignupInputGroup
          iconClassName='fas fa-lock'
          name='confirm password'
          type='password' placeholder='confirm password'
          inValidfeedback=' Please confirm your password.'
        />
    </div>
    <button className="btn continue">Continue</button>
    <div style={loading}>
        <Loader color="#0FC86F" speed={1}className="spinner"/>
    </div>
    </div>
 </form>
);

UpdateAccountForm.propTypes = {
  email: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  loading: PropTypes.object
};

export default UpdateAccountForm;
