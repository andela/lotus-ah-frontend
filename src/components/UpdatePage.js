// react libraries
import React, { Component } from 'react';

// third party libraries
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// helpers
import decodedToken from '../helpers/decodeToken';

// Modules importations
import completeUserRegistration from '../action/signup/update';
import Logo from './reusables/logo/Logo';
import '../styles/_form.scss';
import '../styles/_update.scss';

/**
 * @class CreateArticle
 * @extends {Component}
 * @param {object} event
 */
class UpdatePage extends Component {
  constructor(props) {
    super(props);

    const { token } = this.props.match.params.token;
    const userData = decodedToken(token);

    this.state = {
      email: userData.email,
      password: '',
      fullname: '',
      username: '',
      confirmPassword: '',
      submitted: false,
      token
    };
  }


  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({ submitted: true });
    const {
      email, password, fullname, username,
    } = this.state;
    const { dispatch } = this.props;
    if (email && password && fullname && username) {
      dispatch(completeUserRegistration(this.state));
    }
  }

  render() {
    return (
        <div className="login">
          <div className="container">
            <div className="row auth-h d-flex align-items-center">
              <div className="col-md-6 offset-md-3 text-center login-wrap">
                <Logo />
                <div className="row">
                  <div className="col-md-10 offset-md-1">
                    <div className="form-wrap">
                      <h1 className="form-title">Complete your registration</h1>
                      <form className="form-row" onSubmit={this.handleSubmit}>
                        <div className="col-10 offset-1">
                          <div className="input-wrap">
                            <div className="input-group">
                              <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroupPrepend"><i className="fas fa-user"></i></span>
                              </div>
                              <input type="text" className="form-control" id="validationCustomUsername" name="email" value={this.state.email} onChange={this.handleChange} placeholder="" aria-describedby="inputGroupPrepend" required />
                              <div className="invalid-feedback">
                                Please choose a username.
                              </div>
                            </div>

                            <div className="input-group">
                              <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroupPrepend"><i className="fas fa-lock"></i></span>
                              </div>
                              <input
                                type="text"
                                className="form-control"
                                id="validationCustomUsername"
                                name="fullname"
                                value={this.state.fullname}
                                onChange={this.handleChange}
                                placeholder="full name"
                                aria-describedby="inputGroupPrepend"
                                required
                              />
                              <div className="invalid-feedback">
                                Please enter your full name.
                              </div>
                            </div>

                            <div className="input-group">
                              <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroupPrepend"><i className="fas fa-lock"></i></span>
                              </div>
                              <input type="text" className="form-control" id="validationCustomUsername" name="username" value={this.state.username} onChange={this.handleChange}placeholder="user name" aria-describedby="inputGroupPrepend" required />
                              <div className="invalid-feedback">
                                Please enter a username.
                              </div>
                            </div>

                            <div className="input-group">
                              <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroupPrepend"><i className="fas fa-lock"></i></span>
                              </div>
                              <input type="password" className="form-control" id="validationCustomUsername" name="password" value={this.state.password} onChange={this.handleChange}placeholder="password" aria-describedby="inputGroupPrepend" required />
                              <div className="invalid-feedback">
                                Please enter your password.
                              </div>
                            </div>

                            <div className="input-group">
                              <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroupPrepend"><i className="fas fa-lock"></i></span>
                              </div>
                              <input type="password" className="form-control" id="validationCustomUsername" name="confirmPassword" value={this.state.confirmPassword} onChange={this.handleChange}placeholder="confirm password" aria-describedby="inputGroupPrepend" required />
                              <div className="invalid-feedback">
                                Please confirm your password.
                              </div>
                            </div>
                          </div>

                          <button className="btn continue" type="submit">Continue</button>
                          </div>
                      </form>

                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
      </div>
    );
  }
}

UpdatePage.propTypes = {
  props: PropTypes.object,
  match: PropTypes.object,
  dispatch: PropTypes.object
};

/**
 *
 * @param {object} state
 * @returns {object} updateRegistration
 */
function mapStateToProps(state) {
  const updateRegistration = state.authentication;
  return {
    updateRegistration
  };
}

export default connect(mapStateToProps)(UpdatePage);
