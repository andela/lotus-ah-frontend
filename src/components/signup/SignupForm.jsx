// react libraries
import React, { Component } from 'react';

// third party libraries
import PropTypes from 'prop-types';


// components
import SignupInputGroup from './SignupInputGroup';

/**
 * @class
 * @description renders signup form component
 * @param {object} event
 */
class SignupForm extends Component {
    state = {
      input: '',
      submittted: false,
      responseMessage: 'A verification email will be sent to your mail box'
    };

/**
 * @desc handles input change event
 *@param {object} event
 *@memberof SignupForm
*/
handleChage = (event) => {
  this.setState({
    input: event.target.value,
    submittted: false,
  });
}

/**
 * @desc handles form submit event
 *@param {object} event
 *@memberof SignupForm
*/
  handleSubmit = (event) => {
    event.preventDefault();
    const email = this.state.input;
    this.props.register(email);
    this.setState({
      input: '',
      submittted: true,
    });
  }

  /**
   * @desc sets response message from server
   *@param {object} event
  *@memberof SignupForm
  */
  setResponseMessage = () => {
    if (this.props.signup.registered === true) {
      this.setState({
        responseMessage: ' verification email has been sent to your mail box'
      });
    } else if (this.props.signup.registered === 'failed' && this.props.signup.progress === 'done') {
      this.setState({
        responseMessage: this.props.signup.message
      });
    }
  }

  render = () => (
   <div className="form-wrap pb-5 text-center">
      <form className="form-row" onSubmit={this.handleSubmit}>
        <div className="col-10 offset-1">
            <div className="input-wrap">
                <SignupInputGroup
                  type={'email'}
                  onChange={this.handleChage}
                  iconClassName={'fas fa-envelope'}
                  inputValue={this.state.input}
                />
                <div
                  className={`email-message text-left alert ${this.props.signup.registered === true ? ' alert-success' : ''}`}
                >
                  <p>{ this.state.responseMessage }</p>
                </div>
            </div>
              <button
                className="btn"
                type="submit"
                disabled={this.state.submittted}
              >
                Sign up
              </button>
        </div>
      </form>
    </div>
  );
}

SignupForm.propTypes = {
  signup: PropTypes.object,
  register: PropTypes.func,
};

export default SignupForm;
