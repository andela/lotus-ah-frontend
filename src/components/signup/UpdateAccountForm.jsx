// react libraries
import React from 'react';

// third party libraries
import PropTypes from 'prop-types';

// components
import SignupInputGroup from './SignupInputGroup';

/**
 * @desc renders update user detail form
 * @param {string} email
 * @param {funnction } handleSubmit
 * @return update form
*/
const UpdateAccountForm = ({ email, handleSubmit }) => (
  <form
    className='form-row'
    onSubmit={handleSubmit}
  >
  <div className="col-10 offset-1">
    <div className="input-wrap">
      <SignupInputGroup
        value={email}
        iconClassName='fas fa-envelope'
        name='email'
        type='email'
        placeholder='email'
        inValidfeedback='Please your email'
        disabled={true}
      />
      <SignupInputGroup
        iconClassName='fas fa-user'
        name='fullname'
        type='text'
        placeholder='fullname'
        inValidfeedback='Please enter your full name'
      />
      <SignupInputGroup
        iconClassName='fas fa-at'
        name='username'
        type='text'
        placeholder='user name'
        inValidfeedback='Please enter a username'
      />
        <SignupInputGroup
          iconClassName='fas fa-lock'
          name='password'
          type='password'
          placeholder='password'
          inValidfeedback='Please enter your password.'
        />
         <SignupInputGroup
          iconClassName='fas fa-lock'
          name='confirm password'
          type='password' placeholder='confirm password'
          inValidfeedback=' Please confirm your password.'
        />
    </div>
    <button className="btn continue">Continue</button>
    </div>
 </form>
);

UpdateAccountForm.propTypes = {
  email: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default UpdateAccountForm;
