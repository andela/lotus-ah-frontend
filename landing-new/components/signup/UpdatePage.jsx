// react libraries
import React, { Component } from 'react';

// third party libraries
import PropTypes from 'prop-types';

// third-party libraries
import { Redirect } from 'react-router-dom';

// components
import Logo from '../reusables/logo/Logo.jsx';
import UpdateAccountForm from './UpdateAccountForm';
import Alert from '../reusables/alert/Alert';

// validator
import validator from '../../helpers/validator';

/**
 * @class UpdatePage
 * @param {object} event
 * @desc renders update page
 */
class UpdatePage extends Component {
  state = {
    alert: false,
    alertOpen: false,
    message: '',
    formError: false,
  }
  closeAlert = () => {
    this.setState({
      alert: false,
    });
  }

  componentDidUpdate = (prevprops, prevState) => {
    if(this.props.updateAccount.update === 'failed' && prevState.alert === false){
        this.setState({
          alert: true
        })
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { errors, data } = this.validateFormFields(event);
    if(errors.length <= 0) {
      this.props.updateUser(data);
      this.setState({
        alert: false,
        formError: false,
      })
    } else {
      this.setState({
        message: errors[0].message,
        formError: true,
        alert: true
      });
    }
  }

  validateFormFields = (event) => {
    const data = new FormData(event.target);
    const [firstname, last] = data.get('fullname').split(' ');
    const lastname = last || firstname;
    data.set('firstname', firstname);
    data.set('lastname', lastname);
    data.set('email', this.props.location.verifiedUser.email);
    const errors = validator.validate({
      password: data.get('password'),
      confirmPassword: data.get('confirm password'),
      fullname: data.get('fullname'),
      username: data.get('username'),
    },
    [
      'password',
      'name'
    ]);
    return { errors, data }
  }

  redirectTohomePage = () => (
    <Redirect
    to={{
      pathname: '/',
      user: this.props.updateAccount.user,
      isAuth: this.props.updateAccount.isAuth,
    }} />
  )

  render() {
    const { update, message } = this.props.updateAccount;
    const loading = this.props.updateAccount.progress === 'ongoing' ? { display: "block" } : { display: "none" };
    if (update === true) {
      return this.redirectTohomePage();
    }
    return (
        <div className="login">
        {update === 'failed' || this.state.formError
          && <div className="update-user-error-alert">
            <Alert
              alert="alert-danger"
              isOpen={this.state.alert}
              onClick={this.closeAlert}
              text={!message ? this.state.message : message}
            />
          </div>
        }
          <div className="container">
            <div className="row auth-h d-flex align-items-center">
              <div className="col-md-6 offset-md-3 text-center login-wrap">
                <Logo />
                <div className="row">
                  <div className="col-md-10 offset-md-1">
                    <div className="form-wrap">
                      <h1 className="form-title">Complete your registration</h1>
                        <UpdateAccountForm
                          loading={loading}
                          handleSubmit={this.handleSubmit}
                          email={ this.props.location.verifiedUser.email}
                        />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
      </div>
    );
  }
}

UpdatePage.propTypes = {
  updateAccount: PropTypes.object.isRequired,
  updateUser: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired
};

export default UpdatePage;
