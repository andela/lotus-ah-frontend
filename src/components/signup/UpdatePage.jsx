// react libraries
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// third-party libraries
import { Redirect } from 'react-router-dom';

// components
import Logo from '../reusables/Logo';
import UpdateAccountForm from './UpdateAccountForm';

/**
 * @class UpdatePage
 * @param {object} event
 * @desc renders update page
 */
class UpdatePage extends Component {
  handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.target);
    const [firstname, last] = data.get('fullname').split(' ');
    const lastname = last || firstname;
    data.set('firstname', firstname);
    data.set('lastname', lastname);
    data.set('email', this.props.location.verifiedUser.email);
    this.props.updateUser(data);
  }

  redirectTohomePage = () => (
    <Redirect
    to={{
      pathname: '/',
      user: this.props.updateAccount.user,
      isAuth: this.props.updateAccount.isAuth,
    }} />
  )

  render() {
    if (this.props.updateAccount.update === true) {
      return this.redirectTohomePage();
    }
    return (
        <div className="login">
          <div className="container">
            <div className="row auth-h d-flex align-items-center">
              <div className="col-md-6 offset-md-3 text-center login-wrap">
                <Logo />
                <div className="row">
                  <div className="col-md-10 offset-md-1">
                    <div className="form-wrap">
                      <h1 className="form-title">Complete your registration</h1>
                        <UpdateAccountForm
                          handleSubmit={this.handleSubmit}
                          email={ this.props.location.verifiedUser.email}
                        />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
      </div>
    );
  }
}

UpdatePage.propTypes = {
  updateAccount: PropTypes.object.isRequired,
  updateUser: PropTypes.func.isRequired,
  location: PropTypes.object.isRequired
};

export default UpdatePage;
