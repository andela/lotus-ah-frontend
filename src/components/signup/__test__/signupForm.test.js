// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

// modules importation
import SignupForm from '../SignupForm';
import SignupPage from '../SingupPage';
import UpdateAccountForm from '../UpdateAccountForm';
import UpdatePage from '../UpdatePage';

configure({
  adapter: new Adapter(),
});

// SnapShot Test
describe('Renders the singup page', () => {
  const register = () => {};

  it('should render the signup form', () => {
    const component = shallow(
      <SignupForm
        signup={{
          registered: false,
        }}
        register={register}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the signup page', () => {
    const component = renderer.create(
      <SignupPage
        register={register}
        signup={{
          registered: false,
        }}
      />,
      {
        disableLifecycleMethods: true,
      }
    );
    expect(component).toMatchSnapshot();
  });
});


// SnapShot Test
describe('Renders the update page page', () => {
  const register = () => {};

  it('should render the update form', () => {
    const handleSubmit = () => {};
    const component = shallow(
      <UpdateAccountForm
        signup={{
          registered: false,
        }}
        register={register}
        email='email@gmail.com'
        handleSubmit={
          handleSubmit
        }
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the update page', () => {
    const email = 'email@test.com';
    const updateUser = () => {};
    const component = shallow(
      <UpdatePage
        updateAccount={{
          update: {}
        }}
        location={{
          verifiedUser: {
            email
          }
        }}
        updateUser={
          updateUser
        }
      />,
      {
        disableLifecycleMethods: true,
      }
    );
    expect(component).toMatchSnapshot();
  });
});
