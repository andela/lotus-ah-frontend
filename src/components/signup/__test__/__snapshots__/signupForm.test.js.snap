// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the singup page should render the signup form 1`] = `
<div
  className="form-wrap pb-5 text-center"
>
  <form
    className="form-row"
    onSubmit={[Function]}
  >
    <div
      className="col-10 offset-1"
    >
      <div
        className="input-wrap"
      >
        <SignupInputGroup
          iconClassName="fas fa-envelope"
          inputValue=""
          onChange={[Function]}
          type="email"
        />
        <div
          className="email-message text-left alert "
        >
          <p>
            A verification email will be sent to your mail box
          </p>
          <p />
        </div>
      </div>
      <button
        className="btn"
        disabled={false}
        type="submit"
      >
        Sign up
      </button>
    </div>
  </form>
</div>
`;

exports[`Renders the singup page should render the signup page 1`] = `
<div
  className="sign-up"
>
  <div
    className="container"
  >
    <div
      className="row auth-h d-flex align-items-center"
    >
      <div
        className="col-md-6 offset-md-3 text-center login-wrap"
      >
        <img
          src="/images/logo.png"
        />
        <div
          className="row"
        >
          <div
            className="col-md-8 offset-md-2"
          >
            <div
              className="form-wrap"
            >
              <h1
                className="form-title"
              >
                Sign up
              </h1>
              <div
                className="sign-up-card-wrap text-left"
              >
                <div
                  className="sign-up-card facebook"
                >
                  <a
                    className="d-flex align-items-center"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                    Sign up with 
                    Facebook
                  </a>
                </div>
                <div
                  className="sign-up-card facebook"
                >
                  <a
                    className="d-flex align-items-center"
                  >
                    <i
                      className="fab fa-twitter"
                    />
                    Sign up with 
                    Twitter
                  </a>
                </div>
                <div
                  className="sign-up-card facebook"
                >
                  <a
                    className="d-flex align-items-center"
                  >
                    <i
                      className="fab fa-google"
                    />
                    Sign up with 
                    Google
                  </a>
                </div>
                <div
                  className="sign-up-card facebook"
                >
                  <a
                    className="d-flex align-items-center"
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-envelope"
                    />
                    Sign up with Email
                  </a>
                </div>
              </div>
            </div>
            <div
              className="form-footer"
            >
              <p>
                Already have an account?
                <a
                  href="./login"
                >
                  Login
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders the update page page should render the update form 1`] = `
<form
  className="form-row"
  onSubmit={[Function]}
>
  <div
    className="col-10 offset-1"
  >
    <div
      className="input-wrap"
    >
      <SignupInputGroup
        disabled={true}
        iconClassName="fas fa-envelope"
        inValidfeedback="Please your email"
        name="email"
        placeholder="email"
        type="email"
        value="email@gmail.com"
      />
      <SignupInputGroup
        iconClassName="fas fa-user"
        inValidfeedback="Please enter your full name"
        name="fullname"
        placeholder="fullname"
        type="text"
      />
      <SignupInputGroup
        iconClassName="fas fa-at"
        inValidfeedback="Please enter a username"
        name="username"
        placeholder="user name"
        type="text"
      />
      <SignupInputGroup
        iconClassName="fas fa-lock"
        inValidfeedback="Please enter your password."
        name="password"
        placeholder="password"
        type="password"
      />
      <SignupInputGroup
        iconClassName="fas fa-lock"
        inValidfeedback=" Please confirm your password."
        name="confirm password"
        placeholder="confirm password"
        type="password"
      />
    </div>
    <button
      className="btn continue"
    >
      Continue
    </button>
    <div>
      <Component
        className="spinner"
        color="#0FC86F"
        component="div"
        loadedClassName="loadedContent"
        parentClassName="loader"
        speed={1}
      />
    </div>
  </div>
</form>
`;

exports[`Renders the update page page should render the update page 1`] = `
<div
  className="login"
>
  <div
    className="container"
  >
    <div
      className="row auth-h d-flex align-items-center"
    >
      <div
        className="col-md-6 offset-md-3 text-center login-wrap"
      >
        <Logo />
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-1"
          >
            <div
              className="form-wrap"
            >
              <h1
                className="form-title"
              >
                Complete your registration
              </h1>
              <UpdateAccountForm
                email="email@test.com"
                handleSubmit={[Function]}
                loading={
                  Object {
                    "display": "none",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
